### PR TITLE
Fix Dutch edge cases

### PR DESCRIPTION
**Pull request by mvz**
See #181 

Just a spec for now ...

### CODE REVIEW

**Comment by jlduran** [comment]
If I understand correctly, the separator (`union`) should be `ën` when the unit is 2 or 3, am I right? For example:

```yaml
21: eenentwintig
22: tweeëntwintig
23: drieëntwintig
24: vierentwintig
```

(I think 21 is also wrong?, but unrelated to this bug, can be fixed in another PR).

If so, then we could add that string to the yaml file, for example, in `numbers.nl.yaml`:

```yaml
union: en
union23: ën
```

Then, in `translations/nl.rb` change the `tens_with_ones` method to:

```ruby
[ones(numbers[0]), tens(numbers[1])].join(union(numbers[0]))
```

And override `union` to something like:

```ruby
private

def union(unit)
  return t :union23 if [2, 3].include?(unit)

  t :union
end
```

Would this be OK?

**Comment by mvz** [comment]
@jlduran that seems sensible, thanks!

**Comment by jlduran** [comment]
I'm thinking about also fixing 21 under this PR. I'll change the title to "Fix Dutch edge cases".

If there is another edge case, just push a failing test.

Thank you!

**Comment by mvz** [comment]
21 is perfect as is!

**Comment by jlduran** [comment]
Oh... I am using https://nl.wikipedia.org/wiki/21_(getal) as a reference.

_één_ + _twintig_ = _eenentwintig_. Should I leave it as _éénentwintig_?

**Comment by mvz** [comment]
Ah yes, you are right.

**Comment by jlduran** [comment]
🤔 Should it be OK if I just change _één_ with _een_?

**Comment by jlduran** [comment]
If there are no more additions, I'll be committing this PR over the weekend. And cutting a new gem.

Thank you!

**Comment by mvz** [comment]
I think at least for millions it should stay één.

**Comment by jlduran** [comment]
Could you point to an official site explaining large numbers?

All I could find is:

https://nl.wikipedia.org/wiki/Grote_getallen
https://nl.wikipedia.org/wiki/Biljoen
https://nl.wikipedia.org/wiki/Triljoen

**Comment by mvz** [comment]
It's surprisingly hard to find a clear source. Regarding 21, I used this: https://onzetaal.nl/taaladvies/eendagsvlieg/, which indicates 1 is 'één' when written loose, but 'een' when attached. Generic information about numbers is at https://onzetaal.nl/taaladvies/getallen-uitschrijven/, but it lacks an example for 1 million.

Note, 'een millioen' is also valid Dutch, but it means 'a million', not 'one million'.

**Comment by mvz** [comment]
I think now things are mostly ok. There is one more complication about whether to use 'één' or 'een' when it appears on its own. Unfortunately it seems the rule is basically: If there can be confusion, use 'één', otherwise, 'een' is also fine.

See https://www.vlaanderen.be/taaladvies/een-%C3%A9%C3%A9n

**Comment by jlduran** [comment]
Great! I will also remove the “experimental” status from NL in the README. 

Another thing we can do, don’t know if it’s valid: we leave it as _een_ an add an option you can pass, for instance, `accents: true` and that should change it to _één_ instead. We already do something similar for other languages. (See`apocopated: true` for Spanish in the README)

What do you think?

**Comment by jlduran** [comment]
Actually, let me rephrase my last comment.

I think the PR is good as is. I'll squash it and commit it.

If you think this is a valid option:

```ruby
I18n.with_locale(:nl) { 1_000_000.to_words accents: false }
=> "een miljoen"
```

It could be added to this or another (future) commit.

File: lib/numbers_and_words/strategies/figures_converter/languages/nl.rb [hunk 1]
```diff
@@ -10,16 +10,11 @@ def hundreds_number_to_words
...
-              if figures.number_in_capacity(@current_capacity) == 1
-                megs
-              else
-                capacity_words = words_in_capacity(@current_capacity)
-                capacity_words.empty? ? [] : [capacity_words, megs].join
-              end
-            else
-              super
-            end
+            return super if FiguresArray::THOUSAND_CAPACITY != @current_capacity
```
Comment (mvz) [inline lib/numbers_and_words/strategies/figures_converter/languages/nl.rb:13]: I think this could be reformatted to make it clearer that the main choice is between thousands and not thousands. Maybe a blank line after this one if you want to avoid nesting completely?

**Comment by mvz** [comment]
Yes, I think having an option to let the user decide between 'een' and 'één' for the loose case is a good way to handle this.

File: lib/numbers_and_words/strategies/figures_converter/languages/nl.rb (hunk 1 above)
Comment (mvz) [reply lib/numbers_and_words/strategies/figures_converter/languages/nl.rb:13]: Or, leave the code structure as it was before this PR, and only use a guard clause inside the `if FiguresArray::THOUSAND_CAPACITY == @current_capacity` branch.

```ruby
            if FiguresArray::THOUSAND_CAPACITY == @current_capacity
              return megs if figures.number_in_capacity(@current_capacity) == 1

              capacity_words = words_in_capacity(@current_capacity)
              capacity_words.empty? ? [] : [capacity_words, megs].join
            else
              super
            end
```

Comment (jlduran) [reply lib/numbers_and_words/strategies/figures_converter/languages/nl.rb:13]: Sure, it's a personal preference.

Probably you'll be reviewing any addition to this code, so better fits your preferred style. Feel free to change it to your desired style (as long as it is OK with RuboCop, Coveralls and Code Climate).

**Comment by jlduran** [comment]
Thank you for your contribution!

We should have this code in a new gem soon.

File: lib/numbers_and_words/strategies/figures_converter/languages/nl.rb (hunk 1 above)
Comment (jlduran) [reply lib/numbers_and_words/strategies/figures_converter/languages/nl.rb:13]: There are a few "best practices" or whatever, that RuboCop enforces now and then. Since it helps maintaining our code, we just... comply.

https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/GuardClause

Apparently in Ruby, it is considered a "better" style. I was just future proofing it.

**Review by jlduran — APPROVED** [review]
If there are no further changes, I'll cut a new gem over the weekend. Thank you once again!

**Comment by mvz** [comment]
Thanks, @jlduran. I may take a stab at adding an `accents` option on Friday.

**Review by jlduran — APPROVED** [review]
